configure.self

```diff
@@ -53,10 +53,14 @@ function checks {
     mkl_lib_check "libpthread" "" fail CC "-lpthread" \
                   "#include <pthread.h>"
 
-    if [[ $ENABLE_C11THREADS == "y" ]]; then
+    if [[ $ENABLE_C11THREADS != n ]]; then
+        case "$ENABLE_C11THREADS" in
+            y) local action=fail ;;
+            *) local action=disable ;;
```

Comment:
This has a risk of swallowing typos, e.g., Y or 1, what do you think about using 'd' as the default (which also coincides with disable, which is the default action)?
And setting the default to 'd' in mkl_check if not provided.